src/main/resources/mj_runtime.c

```diff
@@ -17,6 +17,12 @@ int main() {
 }
 
 void* calloc_impl(size_t num, size_t size){
+  //printf("num=%zu size=%zu ", num, size);
+  // the upper 4 bytes may contain rubish
+  // and we don't need to allocate such large memory areas at once
+  // we can still allocate around 4 Gbytes
+  num = num & 0x00000000ffffffff;
+  size = size & 0x00000000ffffffff;
```

Comment:
hmm... maybe we can replace the `size_t`  parameters by `int32_t` (again)? That should make the C compiler do the necessary extensions, I think.